conans/client/installer.py

```diff
@@ -236,6 +237,8 @@ def _build(self, nodes_by_level, skip_private_nodes, deps_graph):
                 log_file = log_file if os.path.exists(log_file) else None
                 log_package_built(package_ref, duration, log_file)
                 self._built_packages.add((conan_ref, package_id))
+                if get_env("CONAN_READ_ONLY_CACHE", False):
```

Comment:
Try to limit the spreading of environment variables passing it to the ConanInstaller constructor from the most possible high level module. manager? conanapi? command?